multiversx/mx-chain-go
# Feat/multiple p2p messengers

## Reasoning behind the pull request
- new network needed to simplify the synchronization process for full archive nodes
  
## Proposed changes
- add new network messenger for this new network

## Testing procedure
- standard system test + specific full archive tests described below
   - start a new seeder with a new protocolID(`seednode/config/p2p.toml`) for the new network(eg. erd/kad/1.0.1)
   - start some nodes as full archive nodes, having the above protocolID on node/config/fullArchiveP2P.toml + add this seeder to InitialPeerList
   - after 20-30 epochs, start a new node in full archive mode, with the above changes as well. This node should easily sync

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
